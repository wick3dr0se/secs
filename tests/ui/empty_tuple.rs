use secs::World;

fn optional_components() {
    let world = World::default();

    world.query(|_, ()| {});
    //~^ ERROR: is not a valid query
}
