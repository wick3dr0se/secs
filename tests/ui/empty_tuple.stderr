error[E0277]: `{closure@tests/ui/empty_tuple.rs:6:17: 6:24}` is not a valid query
   --> tests/ui/empty_tuple.rs:6:17
    |
6   |     world.query(|_, ()| {});
    |           ----- ^^^^^^^^^^
    |           |
    |           required by a bound introduced by this call
    |
    = help: the trait `secs::query::Query<_>` is not implemented for closure `{closure@tests/ui/empty_tuple.rs:6:17: 6:24}`
    = note: only tuples with 1 or up to 5 elements can be used as queries
note: required by a bound in `World::query`
   --> $DIR/src/world.rs
    |
LLL |     pub fn query<Q: Query<T>, T>(&self, f: Q) {
    |                     ^^^^^^^^ required by this bound in `World::query`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
